winthrop/books/forms.py

```diff
@@ -55,28 +137,106 @@ class SearchForm(forms.Form):
     query = forms.CharField(label='Keyword or Phrase', required=False)
 
     sort = forms.ChoiceField(widget=RadioSelectWithDisabled, choices=SORT_CHOICES,
-        required=False)
+                             required=False)
+
+    # Solr facet choice fields
+    author = FacetChoiceField()
+    editor = FacetChoiceField()
+    translator = FacetChoiceField()
+    language = FacetChoiceField()
+    subject = FacetChoiceField()
+    annotator = FacetChoiceField()
+    # range choice fields
+    pub_year = RangeField(
+        label='Publication Year',
+        required=False,
+        widget=RangeWidget(attrs={
+            'size': 4,
+            '_inline': True
+        })
+    )
+    # map solr facet field to corresponding form input
+    solr_facet_fields = {
+        'author_exact': 'author',
+        'editor_exact': 'editor',
+        'translator_exact': 'translator',
+        'language_exact': 'language',
+        'subject_exact': 'subject',
+        'annotator_exact': 'annotator'
+    }
+
+    # TODO: Should this be a dict? Right now it doesn't need a different name
+    # to map to the fields used in the form.
+    range_facets = ['pub_year']
```

Comment:
range fields won't ever have suffixes like "_exact", right? this seems fine to me.